src/integrations/transform.mjs

```
@@ -0,0 +1,194 @@
+ * Walk all nodes in a unist tree, calling visitor(node, index, parent) for each.
+ * The visitor may mutate parent.children; return false to skip node's children.
+ */
+function walk(tree, visitor) {
+  function traverse(node, index, parent) {
+    const result = visitor(node, index, parent)
+    if (result === false) return
+    if (node.children) {
+      for (let i = 0; i < node.children.length; i++) {
+        traverse(node.children[i], i, node)
+        // Re-check index in case children were spliced
+        if (node.children[i] !== undefined && node.children[i] !== node.children[i]) i--
```

Biome (2.4.7)

[error] Comparing to itself is potentially pointless.

(lint/suspicious/noSelfCompare)